packages/blade/src/components/Drawer/Drawer.web.tsx

```diff
@@ -0,0 +1,223 @@
+import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
+import { useId } from '~utils/useId';
+import { useVerifyAllowedChildren } from '~utils/useVerifyAllowedChildren';
+
+const SHOW_DRAWER = 'show-drawer';
+
+const AnimatedDrawerContainer = styled(BaseBox)<{
+  isFirstDrawerInStack: boolean;
+  isVisible: boolean;
+}>(({ theme, isFirstDrawerInStack, isVisible }) => {
+  const entranceTransition: CSSProperties['transition'] = `all ${castWebType(
+    castWebType(makeMotionTime(theme.motion.duration.gentle)),
+  )} ${castWebType(theme.motion.easing.entrance.revealing)}`;
+
+  const exitTransition: CSSProperties['transition'] = `all
+  ${castWebType(makeMotionTime(theme.motion.duration.xmoderate))}
+  ${castWebType(theme.motion.easing.exit.revealing)}`;
+
+  return {
+    opacity: isVisible ? 1 : 0,
+    transform: isVisible
+      ? isFirstDrawerInStack
+        ? 'translateX(calc(-100% - 16px))'
+        : 'translateX(-100%)'
+      : 'translateX(0%)',
+    transition: isVisible ? entranceTransition : exitTransition,
+    animationFillMode: 'initial',
+  };
+});
+
+const DrawerOverlay = styled(FloatingOverlay)(({ theme }) => {
+  return {
+    opacity: 0,
+    transition: `opacity
+      ${makeMotionTime(theme.motion.duration.xmoderate)}
+      ${castWebType(theme.motion.easing.exit.revealing)}`,
+    backgroundColor: theme.colors.overlay.background.subtle,
+
+    [`&.${SHOW_DRAWER}`]: {
+      opacity: 1,
+      transition: `opacity ${makeMotionTime(theme.motion.duration.gentle)} ${castWebType(
+        theme.motion.easing.entrance.revealing,
+      )}`,
+    },
+  };
+});
+
+const _Drawer = ({
+  isOpen,
+  onDismiss,
+  zIndex = componentZIndices.drawer,
+  children,
+  accessibilityLabel,
+  showOverlay = true,
+  initialFocusRef,
+  testID,
+}: DrawerProps): React.ReactElement => {
+  const closeButtonRef = React.useRef<HTMLDivElement>(null);
+  const [zIndexState, setZIndexState] = React.useState<number>(zIndex);
+
+  useVerifyAllowedChildren({
+    children,
+    componentName: 'Drawer',
+    allowedComponents: [drawerComponentIds.DrawerHeader, drawerComponentIds.DrawerBody],
+  });
+
+  const { theme } = useTheme();
+  const drawerId = useId('drawer');
+  const { drawerStack, addToDrawerStack, removeFromDrawerStack } = useDrawerStack();
+
+  const { isMounted, isVisible } = usePresence(isOpen, {
+    enterTransitionDuration: theme.motion.duration.gentle,
+    exitTransitionDuration: theme.motion.duration.xmoderate,
+    initialEnter: true,
+  });
+
+  const { stackingLevel, isFirstDrawerInStack } = React.useMemo(() => {
+    // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+    const level = drawerStack.indexOf(drawerId) + 1;
+    return {
+      stackingLevel: level,
+      isFirstDrawerInStack: level === 1 && drawerStack.length > 1,
+    };
+  }, [drawerId, drawerStack]);
+
+  const { refs, context } = useFloating({
+    open: isMounted,
+  });
+
+  React.useEffect(() => {
```

Comment:
We didn't need layoutEffect?

Reply:
nope. It's not critical for layout because anyway drawer is not visible, by the time it opens with animation the effect runs so its fine